assets/src/state.ts

```diff
@@ -532,9 +561,21 @@ const routeTabsReducer = (
           newRouteTab(state.length),
         ],
       ]
+    case "CREATE_PRESET":
+      return [state, [...state, newPreset(action.payload.sourceTab)]]
+    case "INSTANTIATE_PRESET":
+      return [
+        state,
+        [
+          ...state.map((existingRouteTab) => {
+            return { ...existingRouteTab, isCurrentTab: false }
+          }),
+          tabFromPreset(action.payload.preset, state.length),
```

Comment:
**suggestion:** if you rewrite `isPreset` to instead reference `presetName`, we don't need to duplicate the entry whenever we create or instantiate a preset - you'd only need to create a second entry when the user begins to make unsaved changes to the contents of the preset, thereby necessitating storing different information.

Reply:
I guess semantically to me it doesn't make sense to have the `routeTabs` be one array if saved presets and open tabs are going to be two totally separate subsets of it, rather than overlapping.

Reply:
OK, so, this comment and your other ones below puzzled me for a while--but now I think I have it. You're thinking of a `RouteTab` as something that can serve as an open tab and/or preset at the same time, whereas as you say, in my implementation they're mutually exclusive--a preset is a prototype for a route tab, not a route tab itself.

There are pros and cons to both of these approaches. For instance, the approach I've taken here lets you instantiate a single preset several times easily, and edit each of those independently. Your approach makes it easier to update the tab name on preset save as you mention.

However, it seems like conflating route tabs and presets is already creating confusion for us. I think a better idea would be to explicitly treat presets and tabs as different things, and introduce a third concept called maybe "tab settings". Tab settings would consist of the open routes, ladder toggles, and crowding toggles, but not e.g. the ordering or preset name. Presets have tab settings, and route tabs have tab settings, and when you instantiate a preset you copy the tab settings into the newly created tab, setting the rest of the fields as appropriate.

It would involve reworking our schema, but there'll never be an easier time to do that than now, and I feel like if we continue down this present path we're going to regret it.

Reply:
Okay, having thought this over I like your idea. My hope was that the combined `routeTabs` would be concise while still easy to understand, but apparently that's not so much the case. I think we can create an additional table / thing (I like `routeTabPreset` or something as a name - seems clearer to me than "tab settings" and also mirrors what it's called i nthe UI). There's some risk of duplicated code because a lot of the schema, utility functions, APIs, etc. will be a quite similar, but hopefully it won't feel too boilerplate-y.

Reply:
One thing to keep in mind is that however we implement this, we'll want to precisely nail down the interactions between saved presets, open tabs, and workspaces - for instance, if we're separating the two concepts, should a workspace be a collection of saved presets or open tabs? I'm kind of leaning towards the former but we'd have to think it through.